sigstore/_sign.py

```diff
@@ -63,41 +62,28 @@ def sign(file: BinaryIO, identity_token: str, ctfe_pem: bytes) -> SigningResult:
 
     logger.debug("Generating ephemeral keys...")
     private_key = ec.generate_private_key(ec.SECP384R1())
-    public_key = private_key.public_key()
 
     logger.debug("Retrieving signed certificate...")
     fulcio = FulcioClient()
 
     oidc_identity = Identity(identity_token)
 
-    # Build an X.509 Certificiate Signing Request - not currently supported
-    # builder = (
-    #     x509.CertificateSigningRequestBuilder()
-    #     .subject_name(
-    #         x509.Name(
-    #             [
-    #                 x509.NameAttribute(NameOID.EMAIL_ADDRESS, email_address),
-    #             ]
-    #         )
-    #     )
-    #     .add_extension(
-    #         x509.BasicConstraints(ca=False, path_length=None),
-    #         critical=True,
-    #     )
-    # )
-    # certificate_request = builder.sign(private_key, hashes.SHA256())
-
-    # NOTE: The proof here is not a proof of identity, but solely
-    # a proof that we possess the private key. Fulcio verifies this
-    # signature using the public key we give it and the "subject",
-    # which is a function of one or more claims in the identity token.
-    # An identity token whose issuer is unknown to Fulcio will fail this
-    # test, since Fulcio won't know how to construct the subject.
-    # See: https://github.com/sigstore/fulcio/blob/f6016fd/pkg/challenges/challenges.go#L437-L478
-    signed_proof = private_key.sign(
-        oidc_identity.proof.encode(), ec.ECDSA(hashes.SHA256())
```

Comment:
I might be missing it, but how to we do the proof-of-possession now? Does it all happen internally within `builder`?

Reply:
Yes, the CSR itself is the signed proof, see https://cryptography.io/en/latest/x509/tutorial/#creating-a-certificate-signing-request-csr.